src/Components/EnvironmentPicker/EnvironmentPickerManager.ts

```diff
@@ -0,0 +1,374 @@
+import { AzureResourceTypes, IAzureResource } from '../../Models/Constants';
+import {
+    ContainersLocalStorageKey,
+    EnvironmentsLocalStorageKey,
+    SelectedContainerLocalStorageKey,
+    SelectedEnvironmentLocalStorageKey,
+    StorageAccountsLocalStorageKey
+} from '../../Models/Constants/Constants';
+import {
+    ADTEnvironmentInLocalStorage,
+    ADTSelectedEnvironmentInLocalStorage,
+    StorageAccountsInLocalStorage,
+    StorageAccountToContainersMapping
+} from './EnvironmentPicker.types';
+
+class EnvironmentPickerManager {
+    static displayTextForEnvironment(env: string | IAzureResource) {
+        try {
+            if (env) {
+                const urlObj = new URL(
+                    this.getResourceUrl(
+                        env,
+                        AzureResourceTypes.DigitalTwinInstance
+                    )
+                );
+                return urlObj.hostname.split('.')[0];
+            } else {
+                return null;
+            }
+        } catch (error) {
+            console.log(error.message);
+            return null;
+        }
+    }
+
+    static displayTextForContainer(
+        container: string | IAzureResource,
+        storageAccount: string | IAzureResource
+    ) {
+        if (container && storageAccount) {
+            const containerName = this.getContainerName(container);
+            const urlObj = new URL(
+                this.getResourceUrl(
+                    storageAccount,
+                    AzureResourceTypes.StorageAccount
+                )
+            );
+            return `${urlObj.hostname.split('.')[0]}/${containerName}`;
+        }
+        return null;
+    }
+
+    static getResourceUrl(
+        resource: IAzureResource | string,
+        resourceType: AzureResourceTypes, // always pass this in case the resource is string type
+        parentResource?: IAzureResource | string
+    ) {
+        if (resource) {
+            if (typeof resource === 'string') {
+                // it means the option is manually entered using freeform
+                if (resourceType) {
+                    switch (resourceType) {
+                        case AzureResourceTypes.DigitalTwinInstance:
+                        case AzureResourceTypes.StorageAccount:
+                            return resource;
+                        case AzureResourceTypes.StorageBlobContainer: {
+                            const storageAccountEndpointUrl = EnvironmentPickerManager.getResourceUrl(
```

Comment:
nit: can we be consistent with either `this` or `EnvironmentPickerManager`? They do the same thing but just makes it easier to read when we are consistent.